acceptance/tests/ticket_2455_on_solaris_init_provider_should_start_service_in_own_smf_contract.rb

```diff
@@ -0,0 +1,144 @@
+test_name "(PUP-2455) Service provider should start Solaris init service in its own SMF contract"
+
+sleepy_daemon_script = <<SCRIPT
+#!/usr/bin/env ruby
+while true
+  sleep (2)
+end
+SCRIPT
+
+sleepy_daemon_initscript = <<INITSCRIPT
+#!/bin/sh
+FIXTURESERVICE="/tmp/sleepy_daemon"
+start(){
+    $FIXTURESERVICE &
+}
+
+stop(){
+    FIXTUREPID=`ps -ef | grep "$FIXTURESERVICE" | grep -v grep | awk '{print $2}'`
+    kill -9 $FIXTUREPID
+}
+
+status(){
+    FIXTUREPID=`ps -ef | grep "$FIXTURESERVICE" | grep -v grep | awk '{print $2}'`
+    if [ "x$FIXTUREPID" = "x" ]; then
+      exit 1
+    else
+      exit 0
+    fi
+}
+case "$1" in
+    start)
+        start
+        ;;
+    stop)
+        stop
+        ;;
+    status)
+        status
+        ;;
+esac
+INITSCRIPT
+
+step "Setup fixture service manifest on master"
+
+  testdir = master.tmpdir('solaris_services_in_own_smf_contract')
+
+  test_manifest = <<MANIFEST
+File {
+  ensure => directory,
+  mode => "0750",
+  owner => #{master.puppet['user']},
+  group => #{master.puppet['group']},
+}
+file { '#{testdir}': }
+file { '#{testdir}/environments': }
+file { '#{testdir}/environments/production': }
+file { '#{testdir}/environments/production/manifests': }
+file { '#{testdir}/environments/production/manifests/site.pp':
+  ensure  => file,
+  content => '
+    node default {
+      service { "sleepy_daemon":
+      provider => "init",
+      enable   => "true",
+      ensure   => "running",
+      status   => "/etc/init.d/sleepy_daemon status",
+      }
+    }
+  ',
+}
+MANIFEST
+
+  apply_manifest_on master, test_manifest
+
+step "Start master"
+  
+  if master.is_pe?
+    master['puppetservice'] = 'pe-puppetserver'
+    agent_service           = 'pe-puppet'
+  else
+    master['puppetservice'] = 'puppetserver'
+    agent_service           = 'puppet'
+  end
+ 
+  master_opts = {
+    'main' => {
+    'environmentpath' => "#{testdir}/environments",
+    }
+  }
+
+  with_puppet_running_on master, master_opts, testdir do
+
+    agents.each do |agent|
+
+    fixture_service = 'sleepy_daemon'
+    fixture_service_stop = '/etc/init.d/sleepy_daemon stop'
+
+      arch = on(agent, facter('architecture')).stdout.chomp
+      plat = on(agent, facter('osfamily')).stdout.chomp
+
+      skip_test unless plat == "Solaris"
```

Comment:
nit pick. this value is available from beaker via `agent['platform']` so this can be abbreviated and save the facter call to

    skip_test unless agent['platform'] =~ /solaris/

or you can use the `confine` beaker helper

    confine :to, :platform => 'solaris'